test/users/engin/compiler/nestedRecordField.future

```diff
@@ -0,0 +1,4 @@
+bug: inner record object as a field
+
+when outer is a record it seems to fail whereas when it is class it seems to be
+working fine
```

Comment:
To clarify this future, I'd say "In the case of nested records, the compiler is unable to create an instance of the outer type due to not being able to detect and appropriately respond to the interdependency of the inner and outer type when creating their default methods, and thus making it unable to determine the return type to generate from its default methods"